---
layout: default
title: Projects
---

<div class="container" style="margin-top:120px; margin-bottom:80px;">
  <h2>Current Projects</h2>
  <p>Here are the projects I'm currently working on.</p>
  <br>
  <!-- Nav tabs -->
  <ul class="nav nav-tabs" role="tablist">
    <li class="nav-item">
      <a class="nav-link active" data-toggle="tab" href="#scdeconv">Single Cell Deconvolution</a>
    </li>
    <li class="nav-item">
      <a class="nav-link" data-toggle="tab" href="#neuron">Multi-grid NEURON</a>
    </li>
    <li class="nav-item">
      <a class="nav-link" data-toggle="tab" href="#samah">Marijuana Sentimental Analysis</a>
    </li>
  </ul>

  <!-- Tab panes -->
  <div class="tab-content">
    <div id="scdeconv" class="container tab-pane active"><br>
      <h3>Improve the performance of single cell transcriptomics deconvolution using marker gene selection</h3>
      <div class="row">
        <div class="col-sm-6">
          <img src="assets/images/scdeconv_p1.png" alt="Demo_fig" style="max-width: 100%;">
        </div>
        <div class="col-sm-6">
          Single cell transcriptomics data gives the measurements of gene expressions for each individual cell. In practice, usually we only have bulk RNA-seq data which contains the gene expressions for a mixture of single cells because of its price and efficiency. When making clinical decisions, however, experts care more about the proportion of different cell types which can be easily deduced in single cell RNA data. To solve this issue, deconvolution is a process which takes the single cell profile as input and gives the cell type proportion within this mixture. In this project, we propose a new marker gene selection method and find that it is able to improve the performance of deconvolution.  
        </div>
      </div>
      <p>Basically, I have done the following works:</p>
      <ul>
        <li>Benchmark the performance of deconvolution on bulk-RNA Seq under various conditions, including different deconvolution algorithms, marker selection strategies and pseudo-bulk generation methods. Based on the results, MuSiC is chosen to be the model to use in deconvolution pipeline because its performance and interoperability.</li>
        <li>Write a pipeline with enables a better performance of deconvolution with proper number of marker genes. It supports two different modes: pseudobulk mode which only takes a scRNA data as input and realbulk mode which takes scRNAdata, real bulk data and its ground truth as input. Pseudobulk mode is easier to use whereas realbulk mode will typically gives us a better result.</li>
      </ul>
    </div>
    <div id="neuron" class="container tab-pane fade"><br>
      <h3>Multi-gridding NEURON reaction-diffusion model development</h3>
      <div class="row">
        <div class="col-sm-4">
          <img src="assets/images/neuron_p2.png" alt="Demo_fig1" style="max-width: 100%;">
        </div>
        <div class="col-sm-8">
          <img src="assets/images/neuron_p1.png" alt="Demo_fig2" style="max-width: 100%;">
        </div>
      </div>
      <div>
        <p>
          A variety of dedicated tools have been developed to simulate brain activities. The NEURON simulator is one of the most popular tools, which supports simulating the chemical dynamics and neuron networks simultaneously. In NEURON, reaction-diffusion module (rxd) help simulate the movement within molecules with respect to their concentrations, which is vital in simulating the neural environments and their dynamics on numerical scales.
        </p>
        <p>
          Traditionally, the diffusion is realized by constructing a uniform mesh grids of a fixed distance dx. The diffusion activity is then evaluated by each time step dt. However, in real implementation, some spaces in the neuron cells, such as dendrites and soma, contain a high density of molecules whereas some spaces are sparse. Using a mesh with a fixed dx causes a loss of computation resources. Hence, we want to dynamically allocate different dx for different parts in neuron cells. To realize this, we have to implement multi-gridding reaction diffusion model. 
        </p>
        In this project, I basically have done the following works:
        <ul>
          <li>Working on adding multi-gridding diffusion features for NEURON simulator in python.</li>
          <li>Derived the mathematical equations for multi-gridding reaction-diffusion model.</li>
          <li>Implemented the weakly-connected implicit-Euler diffusion model of neuron simulation with a 1% error</li>
          <li>Utilized machine learning algorithms, including XGBoost, random forest, neural network to predict the spike delay of a specific neuron given its previous spike, estim and istim.</li>
        </ul>
      </div>
    </div>
    <div id="samah" class="container tab-pane fade"><br>
      <h3>Sentimental analysis and trend analysis of marijuana projects</h3>
      <p>Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam.</p>
    </div>
  </div>
</div>
